src/main/java/ai/knowly/langtorch/llm/minimax/MiniMaxService.java

```diff
@@ -0,0 +1,143 @@
+            .setTimeoutDuration(timeout)
+            .build());
+  }
+
+  public static MiniMaxApi buildApi(String groupId, String apiKey, Duration timeout) {
+    ObjectMapper mapper = defaultObjectMapper();
+    OkHttpClient client =
+        buildClient(
+            MiniMaxServiceConfig.builder()
+                .setGroupId(groupId)
+                .setApiKey(apiKey)
+                .setTimeoutDuration(timeout)
+                .build());
+    Retrofit retrofit = defaultRetrofit(client, mapper);
+    return retrofit.create(MiniMaxApi.class);
+  }
+
+  public static Retrofit defaultRetrofit(OkHttpClient client, ObjectMapper mapper) {
+    return new Retrofit.Builder()
+        .baseUrl(BASE_URL)
+        .client(client)
+        .addConverterFactory(JacksonConverterFactory.create(mapper))
+        .addCallAdapterFactory(GuavaCallAdapterFactory.create())
+        .build();
+  }
+
+  public static OkHttpClient buildClient(MiniMaxServiceConfig miniMaxServiceConfig) {
+    OkHttpClient.Builder builder =
+        new OkHttpClient.Builder()
+            .addInterceptor(
+                new MiniMaxAuthenticationInterceptor(
+                    miniMaxServiceConfig.groupId(), miniMaxServiceConfig.apiKey()))
+            .connectionPool(new ConnectionPool(5, 1, TimeUnit.SECONDS))
+            .readTimeout(miniMaxServiceConfig.timeoutDuration().toMillis(), TimeUnit.MILLISECONDS);
+
+    return builder.build();
+  }
+
+  public static ObjectMapper defaultObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+    return mapper;
+  }
+
+  public ChatCompletionResult createChatCompletion(ChatCompletionRequest request) {
+    return execute(miniMaxApi.createChatCompletion(request));
+  }
+
+  public ListenableFuture<ChatCompletionResult> createChatCompletionAsync(
+      ChatCompletionRequest request) {
+    return miniMaxApi.createChatCompletion(request);
+  }
+
+  public EmbeddingResult createEmbeddings(EmbeddingRequest request) {
+    return execute(miniMaxApi.createEmbeddings(request));
+  }
+
+  public ListenableFuture<EmbeddingResult> createEmbeddingsAsync(EmbeddingRequest request) {
+    return miniMaxApi.createEmbeddings(request);
+  }
+
+  /**
+   * Calls the MiniMax AI api, returns the response, and parses error messages if the request fails
+   */
+  public static <T> T execute(ListenableFuture<T> apiCall) {
+    try {
+      return apiCall.get();
+    } catch (InterruptedException e) {
+      // Restore the interrupt status
+      Thread.currentThread().interrupt();
+      // Optionally, log or handle the exception here.
+      logger.atSevere().withCause(e).log("Thread was interrupted during API call.");
+      throw new MiniMaxApiServiceInterruptedException(e);
+
+    } catch (ExecutionException e) {
```

Comment:
The actual exception is buried deeply inside the cause where you can get from e.getCause() and specifically it's a httpException
https://github.com/square/retrofit/blob/69315219bc9068b82a3c320fd6297fed045ece7e/retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java#L48

Please update here according based on:
https://github.com/Knowly-ai/langtorch/blob/55f9b370a883e5a40bc20eaab51f1fee56fcdfb0/src/main/java/ai/knowly/langtorch/llm/openai/OpenAIService.java#LL96C7-L107C8

and please also write a test for this.

Reply:
I missed here，thank you.Minimax does not directly throw an HTTP exception, but carries an error code in the returned body. So I resoled this issue outside.

`
  public ChatCompletionResult createChatCompletion(ChatCompletionRequest request) {
    ChatCompletionResult chatCompletionResult =
        execute(
            futureRetrier.runWithRetries(() -> api.createChatCompletion(request), result -> true));

    checkResp(chatCompletionResult.getBaseResp());
    return chatCompletionResult;
  }

  /** Throw exception messages if the request fails */
  public void checkResp(BaseResp baseResp) {
    if (baseResp.getStatusCode() != 0) {
      throw new MiniMaxApiBusinessErrorException(baseResp.getStatusCode(), baseResp.getStatusMsg());
    }
  }
`